.github/workflows/snap-store-publish-to-candidate.yml

```diff
@@ -78,21 +82,78 @@ jobs:
-      revision_amd64: ${{ steps.publish.outputs.revision_amd64 }}
-      revision_arm64: ${{ steps.publish.outputs.revision_arm64 }}
-      version: ${{ steps.build.outputs.version }}
+
+      - name: Upload snap artifact for testing
+        uses: actions/upload-artifact@v3
+        with:
+          name: snap
+          path: ${{ steps.build.outputs.snap }}
+
+  # Create an issue using the template that asks maintainers to test the snap, and take
+  # action to promote the revisions to stable if testing is successful.
   create_issue:
     name: "Create call for testing"
     needs: build
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
       - uses: JasonEtco/create-an-issue@v2
+        id: comment
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
           revision_amd64: ${{ needs.build.outputs.revision_amd64 }}
           revision_arm64: ${{ needs.build.outputs.revision_arm64 }}
           version: ${{ needs.build.outputs.version }}
         with:
           filename: .github/testing-issue-template.md
+    outputs:
+      issue_number: ${{ steps.comment.outputs.number }}
+
+  # Deploy the snap inside a desktop VM and grab screenshots of the desktop, and of
+  # the application, then post them as a comment on the issue created above.
+  grab-screenshots:
+    name: Gather screenshots
+    needs: 
+      - build
+      - create_issue
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout the code
+        uses: actions/checkout@v4
+
+      - name: Get the snap artifact
+        uses: actions/download-artifact@v3
+        with:
+          name: snap
+
+      - name: Setup LXD
+        uses: canonical/setup-lxd@v0.1.1
+
+      - name: Setup test machine and start Signal
+        id: screenshots
+        run: |
+          .github/vmctl prepare
+          .github/vmctl push-snap "signal-desktop_${{ needs.build.outputs.version }}_amd64.snap"
+          .github/vmctl exec "snap run signal-desktop &>/home/ubuntu/signal.log &"
+          sleep 20
+
+          screenshot_full="$(.github/vmctl screenshot-full)"
+          echo "Screenshot of full screen at: $screenshot_screen"
+          echo "screen=$screenshot_full" >> "$GITHUB_OUTPUT"
+
+          screenshot_window="$(.github/vmctl screenshot-window)"
+          echo "Screenshot of window at: $screenshot_window"
+          echo "window=$screenshot_window" >> "$GITHUB_OUTPUT"
+
+          # Output the Signal logs captured from stdout/stderr (for debgging if required)
+          .github/vmctl exec "cat /home/ubuntu/signal.log"
+
+      - name: Comment on call for testing issue with screenshots
+        uses: GrantBirki/comment@v2.0.8
+        with:
+          issue_number: ${{ needs.create_issue.outputs.issue_number }}
+          body: |
+            The following screenshots were taken during automated testing:
+
+            ![window](${{ steps.screenshots.outputs.window }})
+            ![screen](${{ steps.screenshots.outputs.screen }})
```

Comment:
Shouldn't there be a blank line between window and screen? Otherwise, markdown sees it as a single paragraph, I think.

Reply:
Changed to be sure